### PR TITLE
Bjc   extended bi condition tests

### PR DESCRIPTION
**Pull request by cragun47**
Fixed a couple of problems in the BiConditional tests.  Adds full set of tests for biconditions.

### CODE REVIEW

File: src/main/java/io/github/linuxforhealth/core/expression/condition/SimpleBiCondition.java [hunk 1]
```diff
@@ -49,6 +49,11 @@ public boolean test(Map<String, EvaluationResult> contextVariables) {
       ConditionPredicateEnum condEnum = ConditionPredicateEnum
           .getConditionPredicate(this.conditionOperator, variable1.getIdentifier());
       if (condEnum != null) {
+        // if var2 is a string and must be converted to an integer to test
```
Comment (klwhaley) [inline src/main/java/io/github/linuxforhealth/core/expression/condition/SimpleBiCondition.java:52]: why does only var2 need to be converted? obviously new to this section of code...but should var1 also be converted?

the comment confused me slightly -- we are doing this Only when the condition includes integers... so can we update the comment to clarify that..

File: src/test/java/io/github/linuxforhealth/core/expression/condition/SimpleBiConditionTest.java [hunk 2]
```diff
@@ -11,64 +11,562 @@
...
+    String condition = "$var1 NOT_EQUALS abc";
+    SimpleBiCondition simplecondition = (SimpleBiCondition) ConditionUtil.createCondition(condition);
+    Map<String, EvaluationResult> contextVariables = new HashMap<>();
+
+    assertThat(simplecondition.test(contextVariables)).isFalse();
+  }
+
+  // CONTAINS test one true and one false
+
+  @Test
```
Comment (klwhaley) [inline src/test/java/io/github/linuxforhealth/core/expression/condition/SimpleBiConditionTest.java:118]: In the Ends with , contains, starts with etc tests you are testing a variable and a value, never two variables...should we a mix a few of those in?

**Review by klwhaley — APPROVED** [review]
Approved. A couple minor comments.

File: src/main/java/io/github/linuxforhealth/core/expression/condition/SimpleBiCondition.java (hunk 1 above)
Comment (cragun47) [reply src/main/java/io/github/linuxforhealth/core/expression/condition/SimpleBiCondition.java:52]: Will do.

File: src/test/java/io/github/linuxforhealth/core/expression/condition/SimpleBiConditionTest.java (hunk 2 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/core/expression/condition/SimpleBiConditionTest.java:118]: Yes.  I will add those to a new pull.

**Pull request merged**